### PR TITLE
Add script to generate new vesting positions

### PR DESCRIPTION
**Pull request by fedgiac**
See updated readme for details on the purpose of the changes.

Also worth mentioning that at this point in time the license of the repository from which we take the code is MIT.

### Test Plan

Start with a test csv file, for example [this](https://github.com/cowprotocol/team-cow-allocation/files/10157134/test_vesting.csv).

Run the script as in the readme and see the output in [add_vesting_tx_builder.json](https://github.com/cowprotocol/team-cow-allocation/files/10168063/add_vesting_tx_builder.json.txt)

Then, try to load this file in the [transaction builder app in the team controller context](https://gnosis-safe.io/app/eth:0xca07EaA4253638D286caD71CBcEec11803F2709A/apps?appUrl=https%3A%2F%2Fapps.gnosis-safe.io%2Ftx-builder).

Simulate the transaction and see that the [events](https://dashboard.tenderly.co/public/safe/safe-apps/simulator/857ceb90-f8cf-440d-b61f-6ad36643787c/logs) look alright.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,4 +81,25 @@ After obtaining an Etherscan API key, run:
...
+The script outputs a file that is compatible with the Safe transaction builder app and can be imported in the team controller safe context for execution. 
+
+```sh
+npx hardhat start-vesting --network mainnet --csv ./vesting_positions.csv
+```
+
+The format of the CSV file is as follows:
+
+```csv
+Address,Number of Tokens,Start Date,Duration (days)
```
Comment (bh2smith) [inline README.md:98]: Would it be easier to parse column headers without such "rich" context?

File: src/contracts/AllocationModule.sol [hunk 2]
```diff
@@ -222,7 +222,7 @@ contract AllocationModule {
     /// Given the parameters of a vesting position, computes how much of the total amount has been vested so far.
     /// @param start Timestamp when the vesting position was started.
     /// @param current Timestamp of the point in time when the vested amount should be computed.
-    /// @param duration How long it takes for this vesting position to be fully vested.
+    /// @param duration How long it takes for this ` position to be fully vested.
```
Comment (bh2smith) [inline src/contracts/AllocationModule.sol:225]: Was this an intended change? Looks weird...

Comment (nlordell) [inline src/contracts/AllocationModule.sol:225]: Seems unintentional

File: src/tasks/startVesting.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  allocationModule: string | undefined;
+  outputFilePath: string;
+}
+
+const CSV_HEADERS = {
+  beneficiary: "Address",
+  amount: "Number of Tokens",
+  start: "Start Date",
+  duration: "Duration (days)",
+} as const;
```
Comment (bh2smith) [inline src/tasks/startVesting.ts:31]: This seems pretty specifically rich, I would propose using something more like 

```
address, num_tokens, start_date, duration_days
```

**Review by nlordell — APPROVED** [review]
Changes look good to me.

**Review by bh2smith — APPROVED** [review]
Seems good. Although its a lot of new code to look through with regards to the safe transaction construction.

**Comment by bh2smith** [comment]
We can always simulate transactions anyway!

**Comment by fedgiac** [comment]
Richard [pointed out](https://cowservices.slack.com/archives/C03DVQREAH0/p1670339288927159) that most of the parameters are actually unneeded and we can use a much more simplified version!
I made the needed changes and updated the test plan with the new output.

**Comment by bh2smith** [comment]
That is much simpler!

**Pull request merged**